File: managment_tournoi_backend/src/main/java/ma/youcode/managment_tournoi_backend/security/authDto/AuthenticationResponseDto.java
package ma.youcode.managment_tournoi_backend.security.authDto;

import com.fasterxml.jackson.annotation.JsonProperty;
import ma.youcode.managment_tournoi_backend.entity.AppRole;

import java.util.UUID;

public record AuthenticationResponseDto(
        UUID id,
        String firstName,
        String LastName,
        String username,
        String email,
        AppRole role,
        @JsonProperty("access_token")
        String accessToken,
        @JsonProperty("refresh_token")
        String refreshToken
){
}

File: managment_tournoi_frontend/src/app/auth/dto/AuthenticationResponseDto.ts
import { AppRole } from "src/app/dto/roleDto/AppRole";

export interface AuthenticationResponseDto {
    id: string;
    firstName: string;
    lastName: string;
    username: string;
    email: string;
    role: AppRole;
    accessToken: string;
    refreshToken: string;
}